flask-server/main.py

```diff
@@ -60,22 +60,26 @@ def store_user():
         }
         return response_object, 200
 
-def fetch_users(limit):
-    query = datastore_client.query(kind='user')
-    query.order = ['registered_on']
-
-    users = query.fetch(limit=limit)
-
-    return users
-
 @app.route('/apis/fetch-users', methods=['GET'])
-def show_users():
-    limit = int(request.headers.get('limit'))    
-    users = fetch_users(limit)
-    array = []
-    for user in users:
-        array.append([user['email'], user['username'], user['password_hash'], user['registered_on'], user['deleted']])
-    return jsonify(array)
+def fetch_users():
+    resp_token = decode_auth_token(request.headers.get('Authorization'))
+    if is_valid_instance(resp_token):
+        query = datastore_client.query(kind='user').add_filter('deleted', '=', False)
+        users = list(query.fetch())
+        array = []
+        for user in users:
+            if user.id != str(resp_token):
```

Comment:
remove the str(), the id is an integer. the if statement will always be true and you won't remove the current user.